engine/script/src/script_ddf.cpp

```diff
@@ -578,20 +590,22 @@ namespace dmScript
 
     void PushDDFNoDecoder(lua_State* L, const dmDDF::Descriptor* d, const char* data, bool pointers_are_offsets)
     {
+        DM_PROFILE("PushDDFNoDecoder");
         uintptr_t pointers_offset = 0;
         if (pointers_are_offsets)
             pointers_offset = (uintptr_t) data;
 
-        lua_newtable(L);
-        for (uint32_t i = 0; i < d->m_FieldCount; ++i)
+        uint32_t field_count = d->m_FieldCount;
+        const dmDDF::FieldDescriptor* fields = d->m_Fields;
+
+        lua_createtable(L, 0, field_count);
+        for (uint32_t i = 0; i < field_count; ++i)
         {
-            const dmDDF::FieldDescriptor* f = &d->m_Fields[i];
+            const dmDDF::FieldDescriptor* f = &fields[i];
 
-            lua_pushstring(L, f->m_Name);
-            DDFToLuaValue(L, &d->m_Fields[i], data, pointers_offset);
-            lua_rawset(L, -3);
+            DDFToLuaValue(L, &fields[i], data, pointers_offset);
+            lua_setfield(L, -2, f->m_Name);
```

Comment:
I always prefer "setfield" as it's 1 function call less (and easier to remember imho)